tck/src/test/java/org/eclipse/microprofile/jwt/tck/container/jaxrs/AudValidationBadAudTest.java

```diff
@@ -0,0 +1,127 @@
+import org.eclipse.microprofile.jwt.tck.util.MpJwtTestVersion;
+import org.eclipse.microprofile.jwt.tck.util.TokenUtils;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.arquillian.testng.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.testng.Assert;
+import org.testng.Reporter;
+import org.testng.annotations.Test;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static org.eclipse.microprofile.jwt.tck.TCKConstants.TEST_GROUP_CONFIG;
+
+/**
+ * Validate the handling of the JWT aud claim.
+ *
+ * Validate the aud claim against the {@linkplain Names#AUDIENCES} property is performed, and fails
+ * if the jwt aud does not match the {@linkplain Names#AUDIENCES} property.
+ */
+public class AudValidationBadAudTest extends Arquillian {
+    /**
+     * The base URL for the container under test
+     */
+    @ArquillianResource
+    private URL baseURL;
+
+    /**
+     * The token used by the test
+     */
+    private static String token;
+
+    /**
+     * Create a CDI aware base web application archive that includes an embedded PEM public key
+     * that is included as the mp.jwt.verify.publickey property.
+     * The root url is /
+     * @return the base base web application archive
+     * @throws Exception - on resource failure
+     */
+    @Deployment()
+    public static WebArchive createDeployment() throws Exception {
+        URL publicKey = AudValidationBadAudTest.class.getResource("/publicKey4k.pem");
+
+        PrivateKey privateKey = TokenUtils.readPrivateKey("/privateKey4k.pem");
+        String kid = "publicKey4k";
+        HashMap<String, Long> timeClaims = new HashMap<>();
+        token = TokenUtils.generateTokenString(privateKey, kid, "/TokenAudiences.json", null, timeClaims);
+
+        // Setup the microprofile-config.properties content
+        Properties configProps = new Properties();
+        // Location points to the PEM bundled in the deployment
+        configProps.setProperty(Names.VERIFIER_PUBLIC_KEY_LOCATION, "/publicKey4k.pem");
+        configProps.setProperty(Names.ISSUER, TCKConstants.TEST_ISSUER);
+        configProps.setProperty(Names.AUDIENCES, "notAValidAudience");  // does not match json, should fail
+        StringWriter configSW = new StringWriter();
+        configProps.store(configSW, "AudValidationBadAudTest microprofile-config.properties");
+        StringAsset configAsset = new StringAsset(configSW.toString());
+
+        WebArchive webArchive = ShrinkWrap
+                .create(WebArchive.class, "AudValidationBadAudTest.war")
+                .addAsManifestResource(new StringAsset(MpJwtTestVersion.MPJWT_V_1_1.name()), MpJwtTestVersion.MANIFEST_NAME)
```

Comment:
Just realized this has to be `MpJwtTestVersion.MPJWT_V_1_2` (and in other new tests)

Reply:
fixed